experimental/Schemes/SpaceGerms.jl

```diff
@@ -0,0 +1,274 @@
+  return point_coordinates(inverted_set(OO(X)))
+end
+
+function point(X::SpaceGerm{<:Any, <:Any, <:GermAtGeometricPoint})
+  return prime_ideal(inverted_set(OO(X)))
+end
+
+Oscar.ring(X::AbsSpaceGerm) = OO(X)
+
+function Oscar.ideal(X::AbsSpaceGerm{<:Ring,<:MPolyQuoLocalizedRing})
+    return modulus(OO(X))
+end
+
+function Oscar.ideal(X::AbsSpaceGerm{<:Ring,<:MPolyLocalizedRing})
+    return ideal(OO(X),[zero(OO(X))])
+end
+
+@attr SpaceGerm function ambient_germ(X::AbsSpaceGerm{<:Ring,<:MPolyQuoLocalizedRing})
+    Y,_ = germ_at_point(localized_ring(OO(X)))
+    return Y
+end
+
+@attr SpaceGerm function ambient_germ(X::AbsSpaceGerm{<:Ring,<:MPolyLocalizedRing})
+    return X
+end
+
+############################################################################################################
+# allow user to specify point also as ideal
+############################################################################################################
+
+function _maxideal_to_point(I::MPolyIdeal)
+  R=base_ring(I)
+  o = degrevlex(gens(R))
+  G=groebner_basis(I)
+  LG = leading_ideal(I;ordering=o)
+  dim(LG)==0 || error("Ideal does not describe finite set of points")
+  vd,vbasis = _vdim_hack(LG)
+  vd ==1 || error("Ideal does not describe a single K-point")
+  return [leading_coefficient(normal_form(v,G)) for v in gens(R)]
+end
+
+### _vdim_hack only to be used until vdim of 0-dimensional ideals is implemented properly
+### _vdim_hack assumes for simplicity that 
+###         1. dim(I)=0 has been tested, 
+###         2.I is a monomial ideal
+function _vdim_hack(I::MPolyIdeal)
+  leer=elem_type(base_ring(I))[]
+  R = base_ring(I)
+  if one(R) in I 
+    return 0,leer
+  end
+  G=groebner_basis(I)
+  M = ideal(R,gens(R))
+  result=[R(1)]
+  J = ideal(R,normal_form(gens(M),I))
+  while dim(J) != length(gens(R))
+    Jtemp = leer
+    JN=gens(J)
+    for i in 1:length(JN)
+      if(JN[i]!=R(0))
+        push!(result,JN[i])
+        push!(Jtemp, JN[i])
+      end
+    end
+    J = ideal(R,Jtemp)
+    J = ideal(R,normal_form(gens(J*M),I))
+  end
+  return length(result),result
+end
+
+############################################################################################################
+### constructors
+############################################################################################################
+function SpaceGerm(X::AbsSpec, a::Vector)
+  R = ambient_ring(X)
+  kk = coefficient_ring(R)
+  b = [kk.(v) for v in a]  ## throws an error, if vector entries are not compatible
+  U = MPolyComplementOfKPointIdeal(R,b)
+  Y = Spec(Localization(OO(X), U)[1])
+  return SpaceGerm(Y)
+end
+
+function SpaceGerm(X::AbsSpec, I::MPolyIdeal)
+  R = base_ring(I)
+  R == ambient_ring(X) || error("rings are not compatible")
+  a = _maxideal_to_point(I)
+  Y = SpaceGerm(X,a)
+  return Y
+end
+
+function SpaceGerm(X::AbsSpec, I::MPolyQuoIdeal)
+  A = base_ring(I)
+  A == OO(X) || error("rings seem incompatible")
```

Comment:
```suggestion
  A === OO(X) || error("rings are incompatible")
```

Reply:
They could be two different copies of the same ring. that is why I say 'seem' (to the computer).

Reply:
There are some guidelines on how to write error messages. I don't know if they say anything specific in this case, but it might be a good occasion to ask around. Personally, I don't have a strong opinion on this.

Reply:
In hindsight, this just indicated that the test was not on spot w.r.t. the specification of the function. With the latest changes to the specification this was resolved automatically and the error message was changed accordingly.